File: Tests/ai-fixtures/installed-cli-stub.py
#!/usr/bin/env python3
"""A fake Claude/OpenCode CLI for the installed-provider harness."""

import json
import os
import sys


ROOT = os.environ["TC_INSTALLED_STUB_ROOT"]
COMMAND = os.path.basename(sys.argv[0])


def record(name, value):
    with open(os.path.join(ROOT, name), "a") as handle:
        handle.write(value + "\n")


record(COMMAND + "-args.log", json.dumps(sys.argv[1:]))

if COMMAND == "opencode" and sys.argv[1:3] == ["session", "delete"]:
    record("deleted.log", sys.argv[3])
    raise SystemExit(0)

prompt = sys.stdin.read()
record(COMMAND + "-prompt.log", prompt)
record(COMMAND + "-environment.log", os.environ.get("OPENCODE_CONFIG_CONTENT", ""))

if COMMAND == "opencode":
    print(json.dumps({"type": "step_start", "sessionID": "ses_stub", "part": {}}))
    print(
        json.dumps(
            {
                "type": "text",
                "sessionID": "ses_stub",
                "part": {"text": "OpenCode reply"},
            }
        )
    )
    print(
        json.dumps(
            {
                "type": "step_finish",
                "sessionID": "ses_stub",
                "part": {"tokens": {"input": 9, "output": 2}},
            }
        )
    )
else:
    print(
        json.dumps(
            {
                "type": "stream_event",
                "event": {"delta": {"type": "text_delta", "text": "Claude reply"}},
            }
        )
    )
    print(
        json.dumps(
            {
                "type": "result",
                "is_error": False,
                "usage": {"input_tokens": 8, "output_tokens": 2},
            }
        )
    )

File: Tests/ai-fixtures/installed-cli-stub.js
#!/usr/bin/env node

const fs = require("node:fs");
const path = require("node:path");

const root = process.env.TC_INSTALLED_STUB_ROOT;
const command = path.basename(process.argv[1]);

function record(name, value) {
  fs.appendFileSync(path.join(root, name), value + "\n");
}

record(command + "-args.log", JSON.stringify(process.argv.slice(2)));

if (command === "opencode" && process.argv.slice(2, 4).join(" ") === "session delete") {
  record("deleted.log", process.argv[4]);
  process.exit(0);
}

if (command === "grok" && process.argv.slice(2, 4).join(" ") === "sessions delete") {
  record("grok-deleted.log", process.argv[4]);
  process.exit(0);
}

const promptFile = process.argv.indexOf("--prompt-file");
const prompt = promptFile >= 0 && process.argv[promptFile + 1]
  ? fs.readFileSync(process.argv[promptFile + 1], "utf8")
  : fs.readFileSync(0, "utf8");
record(command + "-prompt.log", prompt);
record(command + "-environment.log", process.env.OPENCODE_CONFIG_CONTENT ?? "");
record(command + "-grok-environment.log", process.env.GROK_DISABLE_AUTOUPDATER ?? "");

if (command === "opencode") {
  console.log(JSON.stringify({ type: "step_start", sessionID: "ses_stub", part: {} }));
  console.log(JSON.stringify({
    type: "text", sessionID: "ses_stub", part: { text: "OpenCode reply" }
  }));
  console.log(JSON.stringify({
    type: "step_finish", sessionID: "ses_stub",
    part: { tokens: { input: 9, output: 2 } }
  }));
} else if (command === "grok") {
  console.log(JSON.stringify({
    type: "system", subtype: "init", session_id: "ses_stub"
  }));
  console.log(JSON.stringify({
    type: "stream_event", session_id: "ses_stub",
    event: { delta: { type: "text_delta", text: "Grok reply" } }
  }));
  console.log(JSON.stringify({
    type: "result", is_error: false, session_id: "ses_stub",
    usage: { input_tokens: 8, output_tokens: 2 }
  }));
} else {
  console.log(JSON.stringify({
    type: "stream_event", event: { delta: { type: "text_delta", text: "Claude reply" } }
  }));
  console.log(JSON.stringify({
    type: "result", is_error: false, usage: { input_tokens: 8, output_tokens: 2 }
  }));
}
